File: tests/python/cases/case_011_publish_profile.py
"""
Task case 011: publish profile payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_011_publish_profile(input_data):
    # Task: publish profile records and prepare transport-ready payload.
    task_name = 'publish_profile'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'profile',
        'stage': 'publish',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_011'] = payload
    return 'ok'

File: tests/nodejs/cases/case_011_publish_profile.js
/**
 * Task case 011: publish_profile
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case011PublishProfile(input) {
  // Task: publish profile records and prepare transport-ready payload.
  const taskName = 'publish_profile';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'profile',
    stage: 'publish',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_011', payload);
  return 'ok';
}

module.exports = {
  case011PublishProfile,
};
